android/src/main/kotlin/com/onegini/mobile/sdk/flutter/helpers/SdkError.kt

```diff
@@ -0,0 +1,56 @@
+package com.onegini.mobile.sdk.flutter.helpers
+
+import com.onegini.mobile.sdk.flutter.OneWelcomeWrapperErrors
+import io.flutter.plugin.common.MethodChannel
+import okhttp3.Response
+
+class SdkError(
+    private val title: String = "Error",
+    private var code: Int = OneWelcomeWrapperErrors.GENERIC.code,
+    private var message: String? = OneWelcomeWrapperErrors.GENERIC.message,
+    private val info: Map<String, Any> = emptyMap(),
+    private val wrapperError: OneWelcomeWrapperErrors? = null,
+    val httpResponse: Response? = null,
+) {
+    private var finalMessage :String = OneWelcomeWrapperErrors.GENERIC.message
+
+    private fun getResponseDetails(): MutableMap<Any, Any> {
+        val response: MutableMap<Any, Any> = mutableMapOf()
+
+        if (httpResponse != null) {
+            response["url"] = httpResponse.request.url.toString()
+            response["statusCode"] = httpResponse.code.toString()
+            response["headers"] = httpResponse.headers.toMap()
+
+            val bodyString = httpResponse.body?.string()
+            if (bodyString != null) {
+                response["body"] = bodyString
+            }
+        }
+
+        return response
+    }
+
+    private fun getDetails(): MutableMap<Any, Any> {
+        val details: MutableMap<Any, Any> = mutableMapOf()
+        details["title"] = title
+        details["code"] = code.toString()
+        details["message"] = finalMessage
+        details["userInfo"] = info
+        details["response"] = getResponseDetails()
```

Comment:
details["response"] is now added on every single error. Should it not only be added on errors involving an http response and should it even be added as a separate property in the first place?

Reply:
reworked; response and info are now only added when specifically set during the SdkError initialisation